packages/hardhat-core/src/internal/core/errors-list.ts

```diff
@@ -237,6 +237,21 @@ Either try using a new directory name, or remove the conflicting files.`,
         "Hardhat attempted to convert the input value to a BigInt, but no known conversion method was applicable to the given value.",
       shouldBeReported: false,
     },
+    CORRUPTED_LOCKFILE: {
+      number: 18,
+      message: `You installed Hardhat with a corrupted lockfile due to the NPM bug #4828.
+
+Please delete your node_modules, package-lock.json, reinstall your project, and try again.`,
```

Comment:
```suggestion
Please delete your node_modules, lockfile (package-lock.json or yarn.lock), reinstall your project, and try again.`,
```

Reply:
Now that I think about it, if this bug is only present in npm and not in yarn, then the current message is correct.